File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    string text = get_string("Text: ");
    // Count letters
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    printf("%i letter(s)\n", letters);
    //

    // Couting words
    int words = 1; // Start at 1 because spaces are in between words.
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    printf("%i word(s)\n", words);
    //

    // Counting sentences
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    printf("%i sentence(s)\n", sentences);
    //

    // Getting L and S
    float l = 100 * (float)letters / (float)words;
    float s = 100 * (float)sentences / (float)words;

    // Plugging in Coleman-Liau formula
    float grade = 0.0588 * l - 0.296 * s - 15.8;

    if (grade < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %.f\n", grade); // %.f prints it without the decimal
}

File: readability.py
# Input user for text.
text = input("Text: ")

letters = 0
word_count = 0
sentence_count = 0
alphabet = [
    'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i',
    'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r',
    's', 't', 'u', 'v', 'w', 'x', 'y', 'z'
]

# Get letter count, ignore spaces and punctuation by using alphabet list.
for letter in text:
    if letter.lower() in alphabet:
        letters += 1

# Get word count.
word_count = len(text.split(' '))

# Get sentence count. There may be a cleaner way to do this, but it works, and in one line of code.
sentence_count = (len(text.split('.')) - 1) + (len(text.split('?')) - 1) + (len(text.split('!')) - 1)

# average number of letters per 100 words in the text
l = 100 * letters / word_count
# average number of sentences per 100 words in the text
s = 100 * sentence_count / word_count

# Coleman-Liau index
index = 0.0588 * l - 0.296 * s - 15.8

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")